investpy/currency_crosses.py

```diff
@@ -0,0 +1,248 @@
+#!/usr/bin/env python
+
+# Copyright 2018-2019 Alvaro Bartolome
+# See LICENSE for details.
+
+import json
+
+import pandas as pd
+import pkg_resources
+
+import requests
+import unidecode
+from lxml.html import fromstring
+
+from investpy import user_agent as ua
+
+
+def retrieve_currency_crosses(test_mode=False):
+    """
+    This function retrieves all the available `currency crosses` indexed on Investing.com, so to
+    retrieve data from them which will be used later for inner functions for data retrieval.
+    All the currency crosses available can be found at: https://es.investing.com/currencies/. Additionally,
+    when currency crosses are retrieved all the meta-information is both returned as a :obj:`pandas.DataFrame`
+    and stored on a CSV file on a package folder containing all the available resources.
+    Note that maybe some of the information contained in the resulting :obj:`pandas.DataFrame` is useless as it is
+    just used for inner function purposes.
+
+    Args:
+        test_mode (:obj:`bool`):
+            variable to avoid time waste on travis-ci since it just needs to test the basics in order to improve code
+            coverage.
+
+    Returns:
+        :obj:`pandas.DataFrame` - currencies:
+            The resulting :obj:`pandas.DataFrame` contains all the currencies meta-information if found, if not, an
+            empty :obj:`pandas.DataFrame` will be returned and no CSV file will be stored.
+
+            In the case that the retrieval process of currencies was successfully completed, the resulting
+            :obj:`pandas.DataFrame` will look like::
+
+                country | name | full_name | tag | id
+                --------|------|-----------|-----|----
+                xxxxxxx | xxxx | xxxxxxxxx | xxx | xx
+
+    Raises:
+        ValueError: raised if any of the introduced arguments is not valid.
+        FileNotFoundError: raised if `index_countries.csv` file does not exists or is empty.
+        ConnectionError: raised if GET requests did not return 200 status code.
+        IndexError: raised if currencies information was unavailable or not found.
+    """
+
+    if not isinstance(test_mode, bool):
+        raise ValueError('ERR#0041: test_mode can just be either True or False')
+
+    results = list()
+
+    head = {
+        "User-Agent": ua.get_random(),
+        "X-Requested-With": "XMLHttpRequest",
+        "Accept": "text/html",
+        "Accept-Encoding": "gzip, deflate, br",
+        "Connection": "keep-alive",
+    }
+
+    # url = "https://www.investing.com/currencies/single-currency-crosses"
+    url = "https://www.investing.com/currencies/streaming-forex-rates-majors"
```

Comment:
This should be properly explained either on the code via comments or opening an issue related to it whenever the pull request is done.